### PR TITLE
tests: Add dns record deletion tests (orphan records)

### PR DESCRIPTION
**Pull request by mikenairn**
Adds/Updates tests to better cover dnsrecord deletion scenarios.

Requires: https://github.com/Kuadrant/kuadrant-operator/pull/941

### CODE REVIEW

File: tests/common/dnspolicy/dnspolicy_controller_test.go [hunk 1]
```diff
@@ -768,41 +802,199 @@ var _ = Describe("DNSPolicy controller", func() {
...
-			// It's not the responsibility of this operator to do this, so we should just check if it's gone
-			// (in case we are running on a cluster that actually has a dnsrecord reconciler running), or that it is marked for deletion
-			//Eventually(func() error { // DNS record removed
-			//	return k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, &kuadrantdnsv1alpha1.DNSRecord{})
-			//}, tests.TimeoutMedium, tests.RetryIntervalMedium).Should(MatchError(ContainSubstring("not found")))
+				//ToDo (mnairn) Theres a bug here, the old gateway target never has these annotations removed
+				//err := k8sClient.Get(ctx, client.ObjectKeyFromObject(gateway), gateway)
+				//g.Expect(err).NotTo(HaveOccurred())
+				//g.Expect(gateway.Annotations).ToNot(HaveKey(v1alpha1.DNSPolicyDirectReferenceAnnotationName))
+				//g.Expect(gateway.Annotations).ToNot(HaveKeyWithValue(v1alpha1.DNSPolicyBackReferenceAnnotationName, policiesBackRefValue))
```
Comment (mikenairn) [inline tests/common/dnspolicy/dnspolicy_controller_test.go:882]: Note all these back references will be removed as part of the SOTW so we don't really need to worry about fixing these. More just here for me to understand what should be happening currently.

File: tests/common/dnspolicy/dnspolicy_controller_test.go [hunk 2]
```diff
@@ -768,41 +802,199 @@ var _ = Describe("DNSPolicy controller", func() {
...
+					ConsistOf(
+						MatchFields(IgnoreExtras, Fields{
+							"Type":   Equal(string(gatewayapiv1alpha2.PolicyConditionAccepted)),
+							"Status": Equal(metav1.ConditionFalse),
+							"Reason": Equal("TargetNotFound"),
+						}),
+					),
+				)
+				//ToDo (mnairn) Theres a bug here, this never gets reset when all records are removed due to target removal
+				//g.Expect(dnsPolicy.Status.TotalRecords).To(Equal(int32(0)))
```
Comment (mikenairn) [inline tests/common/dnspolicy/dnspolicy_controller_test.go:897]: Issues with the record count will be resolved by the SOTW status reconciler updates.

File: tests/common/dnspolicy/dnspolicy_controller_test.go [hunk 3]
```diff
@@ -617,17 +621,227 @@ var _ = Describe("DNSPolicy controller", func() {
...
+
+			By("creating second gateway")
+			Expect(k8sClient.Create(ctx, gateway2)).To(Succeed())
+			Eventually(func(g Gomega) {
+				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(gateway2), gateway2)).To(Succeed())
+				gateway2.Status.Addresses = []gatewayapiv1.GatewayStatusAddress{
+					{
+						Type: ptr.To(gatewayapiv1.IPAddressType),
+						//Using the same address can cause a panic in the inmemory provider see https://github.com/Kuadrant/dns-operator/issues/272
+						Value: "172.1.1.1",
```
Comment (mikenairn) [inline tests/common/dnspolicy/dnspolicy_controller_test.go:783]: https://github.com/Kuadrant/dns-operator/issues/272

**Comment by mikenairn** [comment]
@KevFan I haven't reviewed the the TLS tests, but we should probably make sure we have appropriate test coverage there also.

**Review by maleck13 — APPROVED** [review]
/lgtm

**Pull request merged**